src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -229,7 +229,11 @@ internal static void TryRunGarbageCollection(IInstaller workloadInstaller, IRepo
 
                     if (usingRollback)
                     {
-                        installer.WriteInstallState(sdkFeatureBand, GetInstallStateContents(manifestsToUpdate));
+                        installer.WriteInstallState(_dotnetPath, sdkFeatureBand, GetInstallStateContents(manifestsToUpdate));
+                    }
+                    else
+                    {
+                        installer.DeleteInstallState(_dotnetPath, sdkFeatureBand);
```

Comment:
Now that the install state file can include both the installation mode and the manifest versions, I don't think we want to delete the whole file here.  That would change the installation mode, which we don't want.  Rather, I think we want to remove the manifests from the install state file.